### ISSUE TITLE
Wurst updater - Error report: Connection reset

### ISSUE THREAD

**Issue by CreeperBoy2001** [Wurst-Imperium/Wurst-Client#966]
# Description
Auto-generated error report.

# Stacktrace
```
java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at sun.security.ssl.InputRecord.readFully(Unknown Source)
	at sun.security.ssl.InputRecord.readV3Record(Unknown Source)
	at sun.security.ssl.InputRecord.read(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.readRecord(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.readDataRecord(Unknown Source)
	at sun.security.ssl.AppInputStream.read(Unknown Source)
	at java.io.BufferedInputStream.read1(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	at sun.net.www.MeteredStream.read(Unknown Source)
	at java.io.FilterInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at tk.wurst_client.updater.Main.download(Main.java:194)
	at tk.wurst_client.updater.Main.access$2(Main.java:179)
	at tk.wurst_client.updater.Main$1.run(Main.java:84)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by Alexander01998** [comment]
Did you have a working internet connection at the time?

**Comment by OMGchangeMYusername** [comment]
@Alexander01998 I'll have to tell you that i had a problem with the auto update last time,my net wasn't really good,but it froze all the time,it never closed that "Updating.." even after i closed Wurst and everything!
it was still there,it's 99% because of my net,but can you make it so it asks whether the user wants to update now or not? or make it cancel if Wurst was closed or something like that..
idk if that would happen again,but i had to restart my PC and my router to make it go away.
but after that,i launched Wurst and it updated successfully.

**Comment by Alexander01998** [comment]
> can you make it so it asks whether the user wants to update now or not?

I have removed the confirmation message for a good reason. Over 80% of the users rejected all the updates, kept using ridiculously outdated versions and then complained about all the bugs that I had fixed months ago.

> or make it cancel if Wurst was closed

The Wurst updater has to run as a separate program in order to replace the old Wurst files with the new ones.
A program can neither replace itself while it's running nor can it replace any files that are being used by another running program. So instead, the Wurst updater downloads the new files while Minecraft is still running and installs them once Minecraft is closed.

> i had to restart my PC and my router to make it go away.

This has nothing to do with your router, you just need to stop the Wurst updater task in your task manager.
